src/components/molecules/AssetStatus.tsx

```diff
@@ -0,0 +1,46 @@
+import { DDO, DID } from '@oceanprotocol/lib'
+import axios, { CancelTokenSource } from 'axios'
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useOcean } from '../../providers/Ocean'
+import { isFileValid } from '../../utils/provider'
+import Status from '../atoms/Status'
+import Tooltip from '../atoms/Tooltip'
+import styles from './AssetStatus.module.css'
+
+export default function AssetStatus({ ddo }: { ddo: DDO | DID }): ReactElement {
+  const { ocean } = useOcean()
+  const [fileConnectivity, setFileConnectivity] = useState<boolean>(true)
+
+  useEffect(() => {
+    async function validateAsset(ddo: DDO | DID, source: CancelTokenSource) {
+      if (!ddo) return
+      if (ddo instanceof DID) ddo = await ocean.metadataCache.retrieveDDO(ddo)
+
+      const fileValid = await isFileValid(
```

Comment:
we might have to move the provider call out of here. I also always have this habit of creating one component as combination of visual and functionality but that will lead to multiple calls to the same thing pretty quickly in our case.

Here we will have clashes with #631 which centralizes the provider file check into `AssetActions/index` and passes down results as props. This is a better pattern to prevent multiple provider calls by multiple components. So this component here then can receive that file check object and go from there.

Basically somebody has to refactor this, question is if we can easily do it in here already, or if @claudiaHash will be left having to refactor a lot

Reply:
Agree with your point of centralising the provider file check at `AssetActions/index`.

I can refactor this here right now if the changes at #631 `AssetActions/index` is almost finalised, perhaps I can just copy that over?

Reply:
@kremalicious At `src/components/molecules/FormFields/AssetSelection.tsx` I think we still need to call provider file check each time a user select a different algorithm for compute asset. Unless we are doing the file check at `AssetActions/index` for all algorithm on mount which can be quite taxing on performance if the algorithm list is long. 
What do you think?

![image](https://user-images.githubusercontent.com/39853992/120414292-57c07300-c38c-11eb-8498-19329ae5338a.png)

Reply:
yeah, I see we have created a nice dead end where the proper way would be to make that status part of all AssetSelection asset data but in case of compute it can be overkill. So let's try at least that in download case we only have 1 provider call in tandem with #631

Reply:
What about the connectivity check for compute's algorithm list, can you provide an approach please?